BDArmory/Weapons/Missiles/BDModularGuidance.cs

```diff
@@ -778,10 +820,192 @@ private Vector3 CruiseGuidance()
             return this._guidance.GetDirection(this, TargetPosition, TargetVelocity);
         }
 
+        #region Orbital Modular Missile Guidance
+        // Code contained within this region is adapted from Hatbat, Spartwo and MiffedStarfish's Kerbal Combat Systems Mod https://github.com/Halbann/StockCombatAI/tree/dev/Source/KerbalCombatSystems.
+        // Code is distributed under CC-BY-SA 4.0: https://creativecommons.org/licenses/by-sa/4.0/
+        private Vector3 OrbitalGuidance()
+        {
+            Vector3 orbitalTarget;
+            if (TargetAcquired)
+            {
+                float timeToImpact;
+                Vector3 targetVector = TargetPosition - vessel.CoM;
+                Vector3 relVel = vessel.GetObtVelocity() - TargetVelocity;
+                Vector3 relVelNrm = relVel.normalized;
+                Vector3 interceptVector;
+                float relVelmag = relVel.magnitude;
+
+                // Calculate max accel
+                Vector3 propulsionVector = vessel.transform.InverseTransformDirection(-GetFireVector(engines, rcsThrusters, -vessel.ReferenceTransform.up));
+                float maxThrust = propulsionVector.magnitude;
+                float maxAcceleration = maxThrust / vessel.GetTotalMass();
+
+                if (targetVessel != null)
+                    missileTarget = targetVessel.isMissile;
+
+                if (!missileTarget)
+                {
+                    timeToImpact = BDAMath.SolveTime(targetVector.magnitude, maxAcceleration, Vector3.Dot(relVel, targetVector.normalized));
+                    Vector3 lead = (relVelNrm * -1) * timeToImpact * relVelmag;
+                    interceptVector = (TargetPosition + lead) - vessel.CoM;
+                }
+                else
+                {;
+                    Vector3 acceleration = vessel.ReferenceTransform.up * maxAcceleration;
+
+                    timeToImpact = AIUtils.TimeToCPA(targetVector, relVel, TargetAcceleration - acceleration, 30);
+                    interceptVector = AIUtils.PredictPosition(targetVector, relVel, TargetAcceleration - acceleration * 0.5f, timeToImpact);
+                    interceptVector = interceptVector.normalized;
+                    
+                    if (Vector3.Dot(interceptVector, targetVector.normalized) < 0)
```

Comment:
There's no need to normalize `targetVector` when only checking the sign of the dot product.
Also the normalization is unnecessary on lines 858 and 861 since it gets normalized on line 864 and isn't used afterwards.